### PR TITLE
Deploy a serverless demo after submitting release PR

### PR DESCRIPTION
**Pull request by ltrung**
**Description of changes:**
- Deploy a serverless demo after submiting a release candidate
- Handle a case in generate-version.js to use package.json when the version specified there is newer than the latest tag.
**Testing**

1. Have you successfully run `npm run build:release` locally? Yes
2. How did you test these changes? Manually run the release script and generate-version.js
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it? No
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? No
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: script/generate-version.js [hunk 1]
```diff
@@ -2,41 +2,48 @@
...
+const semver = require('semver');
 const exec = require('child_process').execSync;
 const path = require('path');
 
-function manuallyCreateDescription() {
+function manuallyCreateDescription(version) {
   // Probably not a checkout. This is what you'll get if you use
   // "Download ZIP" in the GitHub UI.
   // Synthesize a version file
   // from the current package.json.
```
Comment (michhyun1) [inline script/generate-version.js:13]: nit: might want to update this

**Review by michhyun1 — DISMISSED** [review]

File: script/release [hunk 2]
```diff
@@ -82,9 +75,6 @@ verbose("npm version #{version_string} --no-git-tag-version")
 
 verbose('git add -A')
 verbose("git commit -m 'Publish #{tag}'")
-verbose("git tag -d #{tag} || true")
-verbose("git tag #{tag}")
-verbose("git tag #{semvertag} -m \"Release #{semvertag}.\"")
```
Comment (richnew10) [inline script/release:87]: Where do we tag our releases after this change?

File: script/release [hunk 3]
```diff
@@ -96,4 +86,7 @@ x = STDIN.gets
 exit(1) unless x.strip == 'yes'
 puts
 
-verbose("git push origin HEAD:release --follow-tags -f")
+verbose("git push origin HEAD:release -f")
+
+Dir.chdir(File.expand_path(File.join(File.dirname(__FILE__), '../demos/serverless')))
+verbose("npm run deploy -- -b chime-sdk-demo-release-testing -s chime-sdk-demo-release-testing")
```
Comment (richnew10) [inline script/release:92]: Deploy to a stack for the relevant version number.

File: script/release (hunk 2 above)
Comment (richnew10) [reply script/release:87]: How can we detect when we are attempting to re-release the same version number? (`npm publish` will fail, but can we detect it earlier?)

**Review by richnew10 — CHANGES_REQUESTED** [review]
You're getting into "change the release process" here, so I'd like to see the documentation or explanation about when version numbers should be changed.

File: script/release (hunk 3 above)
Comment (ltrung) [reply script/release]: Updated

File: script/release [hunk 4]
```diff
@@ -54,20 +54,14 @@ end
 package['version'] = version_string
 changelog_file = 'CHANGELOG.md'
 tag = "amazon-chime-sdk-js@#{version_string}"
```
Comment (richnew10) [inline script/release:56]: Do you still need this?

File: script/release [hunk 5]
```diff
@@ -96,4 +87,7 @@ x = STDIN.gets
 exit(1) unless x.strip == 'yes'
 puts
 
-verbose("git push origin HEAD:release --follow-tags -f")
+verbose("git push origin HEAD:release -f")
```
Comment (richnew10) [inline script/release:90]: This would benefit from a comment.

File: script/generate-version.js [hunk 6]
```diff
@@ -1,42 +1,25 @@
...
+const git = require('git-rev-sync');
 const path = require('path');
 
-function manuallyCreateDescription() {
-  // Probably not a checkout. This is what you'll get if you use
-  // "Download ZIP" in the GitHub UI.
-  // Synthesize a version file
-  // from the current package.json.
+const package = readFileSync(path.join(__dirname, '../package.json'));
+const version = JSON.parse(package).version;
```
Comment (richnew10) [inline script/generate-version.js:8]: I suggest logging this.

**Review by richnew10 — APPROVED** [review]
If this works, rock on, but please do make a note of the process improvements we talked about.

Nits below.

**Pull request merged**